### ISSUE TITLE
Client intermittently loses connection 

### ISSUE THREAD

**Issue by TinLe** [symless/synergy#4949]
Updated to 1.7.4 on both RHEL 6.4 (server) and OSX 10.10.4 (client).   The connection on OSX side get lost quite often.   While using the mouse or keyboard, I'll lose control over OSX.  Checking on OSX side, I see that synergy show it is in "stopped" state.   On OSX side, using its local mouse, clicking on "start" does not fix issue.

On Linux side, I see in the log:

[2015-08-11T10:18:47] WARNING: a client with name "tinle-mn1" is already connected
[2015-08-11T10:18:47] WARNING: a client with name "tinle-mn1" is already connected
[2015-08-11T10:18:47] WARNING: a client with name "tinle-mn1" is already connected
[2015-08-11T10:18:49] WARNING: a client with name "tinle-mn1" is already connected
[2015-08-11T10:18:52] WARNING: a client with name "tinle-mn1" is already connected
[2015-08-11T10:18:55] WARNING: a client with name "tinle-mn1" is already connected
[2015-08-11T10:18:55] WARNING: a client with name "tinle-mn1" is already connected
[2015-08-11T10:18:57] WARNING: a client with name "tinle-mn1" is already connected

If I stop and start on Linux side, then on OSX side, "start", then things work again.   But then this connection will sometime later get into same state while I am using it.

Something I noticed is that this is guarantee to happen whenever I put either server or client into screen lock mode.   Whenever I come back, I have to restart Linux synergy, and then start OSX synergy client.

**Comment by Corona4456** [comment]
I'm seeing similar behavior as well. Ever since RC7, I've been getting randomly disconnected and it won't reconnect unless I restart both the server and the client. I'm not seeing the same logs as TinLe on the server size though. I do see this on the client side:
```
[2015-08-11T12:06:05] NOTE: server is dead
[2015-08-11T12:06:06] WARNING: failed to connect to server: server is not respond
ing
```
Then client appears to restart:
```
[2015-08-11T12:06:18] ERROR: failed to connect secure 
socket
[2015-08-11T12:06:19] NOTE: disconnected from server
2015-08-11 12:06:52.325 synergyc[4037:507] cocoa: release
[2015-08-11T12:06:52] NOTE: stopped client
[2015-08-11T12:06:52] INFO: process exited normally
[2015-08-11T12:06:52] INFO: detected process not running, auto restarting
[2015-08-11T12:06:53] INFO: starting client
[2015-08-11T12:06:53] INFO: config file: /var/folders/33/nksd3_j52lz3lg48bvzp95qmcfpb5m/T/Synergy.ux4013
[2015-08-11T12:06:53] INFO: log level: INFO
[2015-08-11T12:06:53] INFO: drag and drop enabled
[2015-08-11T12:06:53] NOTE: started client
```
Even though client was restarted, I still have restart both the server-side and client-side (as mentioned before).

Client is Mac OS X 10.9 and server side is Linux (Ubuntu 12.04).

**Comment by miraculixx** [comment]
+1 

I see the same happening with the following setup. It usually happens when _leaving_ the client screen, not while the focus is on the client. When the connection drops, there is no need to restart either client or server, all it takes is to click "Apply" on the client and it immediately reconnects. Would be nice if it does that automatically. Without clicking apply the connection usually does not recover, even when waiting several minutes. 

_Client_ 

* Synergy Version 1.7.4
* Linux 3.13 Ubuntu (Linux Mint 17 Qiana)
* this is a laptop connected to an external screen

```
# annotated debug
# -- move cursor to linux
[2015-08-12T12:57:49] INFO: entering screen
# -- move cursor back to osx
[2015-08-12T12:58:01] NOTE: server is dead
[2015-08-12T12:58:01] INFO: leaving screen
[2015-08-12T12:58:01] DEBUG: added format 0 for target STRING (31) (1 byte)
[2015-08-12T12:58:01] DEBUG: close clipboard 1
[2015-08-12T12:58:01] DEBUG: retry in 1 seconds
# apparently can't connect 
[2015-08-12T12:58:01] WARNING: failed to connect to server: server is not responding
[2015-08-12T12:58:02] NOTE: connecting to '192.168.0.8': 192.168.0.8:24800
[2015-08-12T12:58:02] DEBUG: open clipboard 0
[2015-08-12T12:58:02] DEBUG: empty clipboard 0
[2015-08-12T12:58:02] DEBUG: grabbed clipboard 0
[2015-08-12T12:58:02] DEBUG: close clipboard 0
[2015-08-12T12:58:02] DEBUG: open clipboard 1
[2015-08-12T12:58:02] DEBUG: empty clipboard 1
[2015-08-12T12:58:02] DEBUG: grabbed clipboard 1
[2015-08-12T12:58:02] DEBUG: close clipboard 1
# after that without clicking apply it won't reconnect automatically, even when waiting several minutes 
# clicking apply immediately reconnects
```

_Server_

* Synergy Version 1.7.4
* OS X 10.9.5
* this is a mac mini connected to a 2nd screen

_Network_ 

Connection is via WLAN which has occasional latency spikes due to interference (e.g. traceroutes average < 5ms but can spike at > 200ms), however otherwise stable.

**Comment by TinLe** [comment]
Updating.  For my particular issue, it turned out there was a second synergyc client running on OSX.  Despite my stopping the old version before installing the new one.   With two synergy client running, this losing connection happens.

At least for me, this issue no longer happens after killing the extra client.

I still see problems from another issue I reported (laggy cursor), and I will update that one to keep this one clean of unrelated reports.

**Comment by TinLe** [comment]
Spoke too soon.  It just happened to me again.  The client and server seem to get out of sync.  On OSX, the widget show that it has "Stop" and the "Start" highlighted, yet I still have connection with the shared keyboard/mouse from Linux.  But it was very laggy, jerky and finally (after a few minutes) become unusable.

Arg, so annoying (cut-n-paste btw Linux and OSX is not working at moment).

The log on Linux side shows:

[2015-08-13T14:28:08] ERROR: ssl error occurred (system call failure)
[2015-08-13T14:28:08] ERROR: eof violates ssl protocol
[2015-08-13T14:28:08] WARNING: failed to accept secure socket
[2015-08-13T14:28:09] ERROR: ssl error occurred (system call failure)
[2015-08-13T14:28:09] ERROR: eof violates ssl protocol
[2015-08-13T14:28:09] ERROR: ssl error occurred (system call failure)
[2015-08-13T14:28:09] ERROR: eof violates ssl protocol

The fix is to quit Synergy on OSX, restart it and things are somewhat happy again.  Except cut-n-paste is not working.

**Comment by TinLe** [comment]
It turns out the second (multiple) synergyc client running problem is caused by synergy itself on OSX.   It has been happening.

As reported in the other issue (https://github.com/synergy/synergy/issues/4132) that I am still seeing laggy cursor.  This morning I came in to office and as soon as I connected my laptop, I noticed the laggy cursor.

Here's the symptoms I noticed that seem to be tied to this.

1. synergyc (OSX) widget on menu bar always seem to indicate it is not running, e.g. the "Start" is clickable and "Stop" is grayed out.

2. cursor laggy, jumps around.

3. synergys (Linux RHEL 6.4) log shows the following (continuously):
[2015-08-19T07:53:31] WARNING: a client with name "tinle-mn1" is already connected
[2015-08-19T07:53:33] WARNING: a client with name "tinle-mn1" is already connected
[2015-08-19T07:53:36] WARNING: a client with name "tinle-mn1" is already connected
[2015-08-19T07:53:38] WARNING: a client with name "tinle-mn1" is already connected
[2015-08-19T07:53:41] WARNING: a client with name "tinle-mn1" is already connected
[2015-08-19T07:53:43] WARNING: a client with name "tinle-mn1" is already connected
[2015-08-19T07:53:46] WARNING: a client with name "tinle-mn1" is already connected
[2015-08-19T07:53:49] WARNING: a client with name "tinle-mn1" is already connected
[2015-08-19T07:53:51] WARNING: a client with name "tinle-mn1" is already connected

4. synergyc (OSX) side, "ps aux |grep synergyc" show multiple synergyc running.  As recent as yesterday, I made sure that there was only one synergyc process running.  So the "7:47AM" process was started this morning by Synergy itself when I came into office.

tinle $ ps axu|grep -i syner
tinle           65549   9.1  0.2   725476  13516   ??  S     7:47AM   0:11.11 /Applications/Synergy.app/Contents/MacOS/synergyc -f --no-tray --debug WARNING --name tinle-mn1 --enable-drag-drop --enable-crypto tinle-ld1:24800
tinle           15859   2.3  0.1   725664   8160   ??  S    12:23PM   4:57.08 /Applications/Synergy.app/Contents/MacOS/synergyc -f --no-tray --debug WARNING --name tinle-mn1 --enable-drag-drop --enable-crypto tinle-ld1:24800
tinle           15850   0.3  0.3  2704976  23776   ??  S    12:23PM   5:38.18 /Applications/Synergy.app/Contents/MacOS/Synergy
tinle           65649   0.0  0.0  2451204    700 s004  S+    7:47AM   0:00.00 grep -i syner

5. If I kill the extra process, then the lag/jerky/jumping mouse/cursor problem go away.

**Comment by miraculixx** [comment]
As per my above [comment](https://github.com/synergy/synergy/issues/4949#issuecomment-130560676) the problem still persists. I have since changed to an ethernet-connected setup where both client and server are connected on the same hub, so no intermittent network issues anymore. The synergy client  however still loses the connection intermittently (and about twice or so an hour) without automatically reconnecting anymore. This gets annoying very quickly... and ideas?

**Comment by Corona4456** [comment]
Just an update, my connection stability improves if I disable encryption. Not ideal but my connection drops very rarely so I'm risking it for now. Can any of the devs comment on this?

**Comment by mattwilmott** [comment]
I've been getting this issue lately and I recently turned on encryption myself. Im running fedora<->fedora in my case. I'll try disabling encryption and try again. 

It looked to me as if there was issues updating the clipboard when leaving the client machine, thats all I could see from the logs. Either way clicking apply again on the client resolved it as mentioned higher up

**Comment by gabrielhaase** [comment]
@mattwilmott
Are you right!! I'm using Mint <-> Mint and the issue goes when the clipboard is updated.

**Comment by sidewinder94** [comment]
Using Windows as a server Ubuntu as a client and got the same problem too.

Some times, however, rather than bluntly disconnecting and saying the server is dead, the client is havin g serious mouse lag (there is no network issue, since as soon as I relaunche the connection from the client, the problem disappear)

Nothings appears in these cases. (Debug level : INFO, didn't go into debug yet)

**Comment by miraculixx** [comment]
@nbolton would like to draw your attention to this issue - is any resolve planned? thanks.

**Comment by sidewinder94** [comment]
Just wanted to try out something but i don't even know if it is possible.....
As I wrote earlier, the shared clipboard seems to have problems before an issue appears (in my case at least).

I'd like to try to disable the shared clipboard to see if there are any improvements..... but i couldn't find the option in the GUI and don't know where the documentation for the Synergy config files is.

**Comment by miraculixx** [comment]
> As I wrote earlier, the shared clipboard seems to have problems before an issue appears (in my case at least).

I've seen this on occasion, too -- when copying text at the synergy client (e.g. from browser or a text editor), then move the cursor to the server, the connection is dropped most of the times. This seems to happen only after the connection was stable for some time and moving back and forth was no problem. Then when you least expect it, on the next copy / move cursor to server - it fails.

However that's not the only cause of this issue. Mind you this has only started to appear since 1.7.4 got out, synergy used to work previously without any issue for many years.

**Comment by falz** [comment]
Just wanted to chime in as another user that experiences this issue. synergy server is OSX, client is linux. This is my daily desktop environment at work. I was previously using 1.6x unregsitered. when I registered/paid a month or so ago I began using 1.7.4. since then, multiple times per day I lose connection. It always seems to be I'm moving my mouse from the server (mac) to the client (linux). The only fix i've found is to go to the client and stop/start it again.

**Comment by TinLe** [comment]
@falz yeah, that was my workaround.   Restart client usually fixes it.   Annoying as hell though.

My setup is reverse of yours.  Linux desktop server and OSX laptop as client.    I upgraded OSX to El Cap 10.11, still see same problem.

**Comment by sidewinder94** [comment]
Seems that the common point between all of the setups is a Linux Client/Server.

Could someone try an OSX<->Windows Setup ? (I would but do not own an OSX device)

**Comment by faerloev** [comment]
I have the exact same problem between 2 Linux boxes (both Gentoo Linux and Synergy 1.7.4), so it's definitely not limited to Linux<->OSX combinations.
I can restore connection by stopping and restarting the server.

**Comment by miraculixx** [comment]
+1 @XinyuHou for giving this bug status. how can we help?

**Comment by XinyuHou** [comment]
@miraculixx 
We suspect this problem is related to clipboard issue.

Here is a pull request about some clipboard fixes.

http://synergy-project.org/nightly?filter=nyetwurk-master-alpha-3eaefc

Please try it out.

**Comment by miraculixx** [comment]
@XinyuHou thank you :+1:  Have installed and used this version (1.8.0-alpha-3eaefcf) for the last few days and since did not experience _any connection loss_. Note I only installed the Linux version only on the Linux client, kept the official 1.7.4 version on the OSX server.

**Comment by TinLe** [comment]
I've been using synergy-v1.7.5-rc1-ceab3a8 on OSX 10.10.5 9 (client laptop) and RHEL 6.5 (desktop server) almost a week now.  Seem to be more stable.  I have seen the lost connection and multiple synergyc processes once each, previously it was at least once a day.

Getting there....

:+1:

**Comment by XinyuHou** [comment]
Hi Guys, please try out our 1.7.5 release to see if it fixes this problem.

**Comment by miraculixx** [comment]
@XinyuHou could not install 1.7.5 yet, however came across this issue in client 1.8.0-alpha-3eaefcf:

```
[2015-12-02T16:55:33] WARNING: timed out 0.500111s waiting for clipboard fill
[2015-12-02T16:56:08] NOTE: server is dead
```

Like before, in this instance the synergy client stops and can only be restarted by local mouse/keyboard.

This happened after weeks of operation without problem (with several client & server restarts in between unrelated to synergy). I suspect it was because of clipboard size (it was some 70KB of text -- ). I guess this is an uncommon edge case, however it would still be nice the client survives this or restarts without physical intervention.

Client 1.8.0-alpha-3eaefcf (Linux)
Server 1.7.5 (OSX 10.11.1)

**Comment by XinyuHou** [comment]
@miraculixx 

Please try this nightly.
http://synergy-project.org/nightly?filter=synergy-v1.7.6-rc1-09d0ec0

**Comment by falz** [comment]
I installed that nightly this morning (mac+linux) and have not had issues yet today. Normal days I would have >1.

**Comment by uhtred** [comment]
Installed synergy-v1.7.6-rc1-5287c62-Linux-x86_64.deb, same problem, update clipboard and stop working.

**Comment by XinyuHou** [comment]
@falz 
Did you mean nightly synergy-v1.7.6-rc1-09d0ec0?

**Comment by XinyuHou** [comment]
@uhtred 
Did you install 5287c62 on both your server and client machines?

**Comment by vinadoros** [comment]
@XinyuHou I was having problems with the synergy client seizing up as well. I have now compiled 09d0ec0 from git and installed on my two linux machines, and for the past 5 hours, I haven't seen any client lose connection yet (the server is a Windows machine running 7a207b4 from master). I will post if I still come across an issue, but this seems to be working for me. Any chance this fix will be pushed to master in the near future?

**Comment by XinyuHou** [comment]
@vinadoros 

We are still working on 1.7.6 and there are some issues in the milestone to be fixed. As soon as we release 1.7.6 we will merge the fixes into master.

**Comment by 99Percent** [comment]
v1.7.5 Xubuntu 15.04 on both server and client connected via LAN. Intermittent "Server is dead". Disabled SSL to no avail. My LAN is rock solid, so no reason for them to disconnect. Corrects by simply clicking "Apply" on the client GUI. How hard is it to implement an auto retry to reconnect method?

**Comment by TinLe** [comment]
On released v1.7.6 and still seeing the occasional lost sync btw client and server.   On the server side, I am seeing this in the log:

[2016-04-04T09:38:35] ERROR: corrupted clipboard data, expected size=88 actual size=0


Server: Synergy v1.7.6, RHEL 6.6
Client: Synergy v1.7.6, OSX 10.11.4

**Comment by Fuco1** [comment]
I run mint <-> mint on latest master (compiled on both machines manually) and also experience this issue.  In addition I use `xcmenu` which is a clipboard/selection manager, maybe that can be causing additional problems? :O Anyone else uses something like that?

**Comment by larpo1** [comment]
Yep - getting this too. Can make any connection between two macosx machines. Get:

WARNING: a client with name "larpo" is already connected

**Comment by XinyuHou** [comment]
Duplicates to #4768

**Comment by JMAN99** [comment]
After several weeks of struggling with this issue, I was finally able to resolve it with this:
- ssh -L 24800:localhost:24800 user@ssh_server

Here's the source page where I found the solution...
https://blog.toloughlin.com/?p=192
